lib/core/src/display/navigation/navigator.rs

```diff
@@ -62,9 +65,21 @@ impl Navigator {
         let drag               = DragProperties::new(1500.0);
         let properties         = PhysicsProperties::new(kinematics, spring, drag);
         let steps_per_second   = 60.0;
-        let properties_clone   = properties.clone();
-        let callback           = move |position| camera.set_position(position);
-        let sim = PhysicsSimulator::new(steps_per_second,properties_clone,callback);
+        let callback           = enclose!((properties) move |interpolated_position:Vector3<f32>| {
+            let fixed_point       = properties.spring().fixed_point;
+            let position          = properties.kinematics().position();
+            let distance          = (position - fixed_point).magnitude();
+            const THRESHOLD : f32 = 1.0;
+            let position = if distance < THRESHOLD {
+                fixed_point
+            } else {
+                interpolated_position
+            };
+            if camera.transform().position() != position {
```

Comment:
Why moving camera has some internal `if` ? What this code is about? And what is this threshold? Basically:

I believe that moving a camera should not need branching in logic - if I'm wrong, I'd love to talk about why and then the code needs to be described better because now it's not clear.

Reply:
@wdanilo there is a good reason for that: we want to avoid the camera to get updated on every frame if its position hasn't changed. We use the threshold to move the camera directly to its final destination (the spring's fixed point), because the position coming from the simulator is unstable, so we can't simply use this comparison `camera.transform().position() != position`.

This is important because we want to stop the simulation by setting the kinematics position and the spring fixed point from the camera position after `set_screen`.

Reply:
@notdanilo how the position coming from simulator could be unstable? 2 follow up questions:
1. If the position coming from simulator is unstable, who's responsibility is it to fix it? Camera?
2. Why it was earlier working, and now it is not? What made the simulator unstable now?